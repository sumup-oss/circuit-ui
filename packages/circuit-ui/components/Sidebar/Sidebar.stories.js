/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { useState } from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { Home, Profile, Sales, Shop } from '@sumup/icons';

import Button from '../Button';

import { Separator } from './components/Separator';

import Sidebar from '.';

export default {
  title: 'Navigation/Sidebar',
  component: Sidebar,
  subcomponents: {
    Header: Sidebar.Header,
    NavList: Sidebar.NavList,
    NavItem: Sidebar.NavItem,
    Aggregator: Sidebar.Aggregator,
    Separator: Sidebar.Separator,
    Footer: Sidebar.Footer,
  },
  parameters: {
    layout: 'fullscreen',
  },
};

const Viewport = styled.div`
  min-height: 500px;
`;

export const Base = (args) => {
  const [open, setOpen] = useState(true);
  const [selected, setSelected] = useState(1);

  return (
    <Viewport>
      {!open && (
        <Button
          onClick={() => setOpen(true)}
          css={css`
            position: absolute;
          `}
        >
          Open sidebar
        </Button>
      )}

      <Sidebar {...args} open={open} onClose={() => setOpen(false)}>
        <Sidebar.Header>Header</Sidebar.Header>
        <Sidebar.NavList>
          <Sidebar.NavItem
            key="home"
            label="Home"
            selected={selected === 1}
            onClick={() => setSelected(1)}
            defaultIcon={<Home />}
            selectedIcon={<Home />}
            tracking={{ label: 'trackingId-home' }}
          />
          <Sidebar.Aggregator
            key="list"
            selected={selected === 2}
            label="List"
            defaultIcon={<Sales />}
            selectedIcon={<Sales />}
          >
            <Sidebar.NavItem
              label="First"
              selected={selected === 21}
              onClick={() => setSelected(21)}
            />
            <Sidebar.NavItem
              label="Second"
              selected={selected === 22}
              onClick={() => setSelected(22)}
            />
            <Sidebar.NavItem
              label="Third"
              selected={selected === 23}
              onClick={() => setSelected(23)}
            />
          </Sidebar.Aggregator>
          <Sidebar.NavItem
            key="shop"
            label="Shop"
            disabled
            selected={selected === 3}
            defaultIcon={<Shop />}
            selectedIcon={<Shop />}
            onClick={() => setSelected(3)}
          />
          <Separator key="separator" />
          <Sidebar.NavItem
            key="me"
            label="Me"
            selected={selected === 4}
            defaultIcon={<Profile />}
            selectedIcon={<Profile />}
            onClick={() => setSelected(4)}
          />
        </Sidebar.NavList>
        <Sidebar.Footer>Footer</Sidebar.Footer>
      </Sidebar>
    </Viewport>
  );
};

Base.args = {
  closeButtonLabel: 'close-button',
};
