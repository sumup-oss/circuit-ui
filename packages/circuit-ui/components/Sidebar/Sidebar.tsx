/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Fragment, ReactNode } from 'react';
import { css } from '@emotion/react';

import styled, { StyleProps } from '../../styles/styled.js';

import { Aggregator } from './components/Aggregator/index.js';
import { Backdrop } from './components/Backdrop/index.js';
import { CloseButton } from './components/CloseButton/index.js';
import { Footer } from './components/Footer/index.js';
import { Header } from './components/Header/index.js';
import { NavItem } from './components/NavItem/index.js';
import { NavList } from './components/NavList/index.js';
import { Separator } from './components/Separator/index.js';

export interface SidebarProps {
  /**
   * The children component passed to the Sidebar
   */
  children: ReactNode;
  /**
   * Determines whether the Sidebar is open
   */
  open: boolean;
  /**
   * Accessibility label for the CloseButton
   */
  closeButtonLabel: string;
  /**
   * A function to handle the sidebar close
   */
  onClose: () => void;
}

const SIDEBAR_WIDTH = 256;

const baseStyles = ({ theme }: StyleProps) => css`
  display: flex;
  flex-direction: column;
  height: 100%;
  min-width: ${SIDEBAR_WIDTH}px;
  background-color: ${theme.colors.n900};
  transition: transform ${theme.transitions.default};
  position: absolute;
  transform: translateX(-100%);
  z-index: ${theme.zIndex.navigation};
  ${theme.mq.giga} {
    transform: translateX(0);
    position: relative;
  }
`;

const openStyles = ({ theme, isOpen }: StyleProps & { isOpen: boolean }) =>
  isOpen &&
  css`
    transform: translateX(0);
    ${theme.mq.giga} {
      transform: translateX(0);
    }
  `;

const Drawer = styled('nav')(baseStyles, openStyles);

/**
 * @legacy
 */
export function Sidebar({
  children,
  open,
  closeButtonLabel,
  onClose,
  ...props
}: SidebarProps) {
  return (
    <Fragment>
      <Drawer isOpen={open} {...props}>
        {children}
      </Drawer>
      <Backdrop
        visible={open}
        onClick={onClose}
        data-testid="sidebar-backdrop"
      />
      <CloseButton
        visible={open}
        label={closeButtonLabel}
        onClick={onClose}
        data-testid="sidebar-close-button"
      />
    </Fragment>
  );
}

/**
 * @legacy
 */
Sidebar.Header = Header;
/**
 * @legacy
 */
Sidebar.Footer = Footer;
/**
 * @legacy
 */
Sidebar.NavList = NavList;
/**
 * @legacy
 */
Sidebar.NavItem = NavItem;
/**
 * @legacy
 */
Sidebar.Aggregator = Aggregator;
/**
 * @legacy
 */
Sidebar.Separator = Separator;
