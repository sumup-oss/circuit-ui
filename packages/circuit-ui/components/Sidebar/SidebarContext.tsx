/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { createContext, ReactNode, useState } from 'react';

export const SidebarContext = createContext({
  isSidebarOpen: false,
  toggleSidebar: () => {},
});

export const SidebarContextConsumer = SidebarContext.Consumer;

export interface SidebarContextProviderProps {
  /**
   * The children received by SidebarContextProvider
   */
  children: ReactNode;
}

export function SidebarContextProvider({
  children,
}: SidebarContextProviderProps) {
  const [isSidebarOpen, setSidebarOpen] = useState(false);

  const toggleSidebar = () => {
    setSidebarOpen((prevState) => !prevState);
  };
  return (
    <SidebarContext.Provider
      value={{
        isSidebarOpen,
        toggleSidebar,
      }}
    >
      {children}
    </SidebarContext.Provider>
  );
}
