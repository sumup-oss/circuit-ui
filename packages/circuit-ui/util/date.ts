/**
 * Copyright 2024, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Temporal } from 'temporal-polyfill';
import { formatDateTime } from '@sumup-oss/intl';

import type { Locale } from './i18n.js';

export type FirstDayOfWeek = 1 | 7;
export type DaysInWeek = number;
export type PlainDateRange =
  | []
  | [Temporal.PlainDate]
  | [Temporal.PlainDate, Temporal.PlainDate];

// ISO 8601 timestamps only support positive 4-digit years
export const MIN_YEAR = 1;
export const MAX_YEAR = 9999;

export const MIN_MONTH = 1;
export const MAX_MONTH = 12;

export const MIN_DAY = 1;
// MAX_DAY is not constant as it depends on the year and month

export function getTodaysDate() {
  return Temporal.Now.plainDateISO();
}

export function isPlainDate(date: unknown): date is Temporal.PlainDate {
  return date instanceof Temporal.PlainDate;
}

export function toPlainDate(date?: string): Temporal.PlainDate | undefined {
  if (!date) {
    return undefined;
  }
  try {
    return Temporal.PlainDate.from(date);
  } catch {
    return undefined;
  }
}

export function sortDateRange<T extends PlainDateRange>(dateRange: T): T {
  return dateRange.sort((a, b) => Temporal.PlainDate.compare(a, b)) as T;
}

export function clampDate(
  date: Temporal.PlainDate,
  minDate?: Temporal.PlainDate | null,
  maxDate?: Temporal.PlainDate | null,
): Temporal.PlainDate {
  if (minDate && Temporal.PlainDate.compare(date, minDate) < 0) {
    return minDate;
  }
  if (maxDate && Temporal.PlainDate.compare(date, maxDate) > 0) {
    return maxDate;
  }
  return date;
}

export function getFirstDateOfWeek(
  date: Temporal.PlainDate,
  firstDayOfWeek: FirstDayOfWeek,
): Temporal.PlainDate {
  const diff =
    (date.dayOfWeek < firstDayOfWeek ? date.daysInWeek : 0) +
    date.dayOfWeek -
    firstDayOfWeek;
  return date.subtract({ days: diff });
}

export function getLastDateOfWeek(
  date: Temporal.PlainDate,
  firstDayOfWeek: FirstDayOfWeek,
): Temporal.PlainDate {
  return getFirstDateOfWeek(date, firstDayOfWeek).add({
    days: date.daysInWeek - 1,
  });
}

export function getMonthName(month: number, locale?: Locale) {
  // The year can be arbitrary since the month names are the same every year
  const yearMonth = new Temporal.PlainYearMonth(2000, month, 'gregory');
  return formatDateTime(yearMonth, locale, { month: 'long' });
}
