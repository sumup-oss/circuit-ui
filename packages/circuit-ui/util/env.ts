/**
 * Copyright 2024, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export function getEnvVariable(name: string) {
  if (name === 'NODE_ENV') {
    // Some bundlers have special logic for `process.env.NODE_ENV` which
    // relies on it being written as a continuous string. Destructuring or
    // dynamically accessing `NODE_ENV` can break this logic.
    throw new Error('Do not dynamically access NODE_ENV');
  }

  if (typeof process !== 'undefined') {
    return process.env?.[name];
  }

  return undefined;
}
