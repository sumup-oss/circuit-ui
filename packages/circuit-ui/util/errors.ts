/**
 * Copyright 2022, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* eslint-disable max-classes-per-file */
import React from 'react';

export class CircuitError extends Error {
  constructor(componentName: string, message: string) {
    super(`[${componentName}] ${message}`);
    this.name = 'CircuitError';
    // Adapted from https://stackoverflow.com/questions/33474179/react-access-parent-component-name

    /* eslint-disable max-len, no-underscore-dangle, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call */
    if (
      // @ts-expect-error Since this code only runs in development, it's fine to use this internal React API.
      React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED
        .ReactCurrentOwner &&
      // @ts-expect-error This is fine.
      React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner
        .return &&
      // @ts-expect-error This is fine.
      React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner
        .return.type
    ) {
      this.stack =
        // @ts-expect-error This is fine.
        React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED
          .ReactCurrentOwner.return.type as string;
    }
    if (
      // @ts-expect-error Since this code only runs in development, it's fine to use this internal React API.
      React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED
        .ReactDebugCurrentFrame &&
      // @ts-expect-error This is fine.
      React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED
        .ReactDebugCurrentFrame.getCurrentStack
    ) {
      this.stack =
        // @ts-expect-error This is fine.
        React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactDebugCurrentFrame.getCurrentStack() as string;
    }
    /* eslint-enable max-len, no-underscore-dangle, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call */
  }
}

export class DeprecationError extends CircuitError {
  constructor(componentName: string, message: string) {
    super(componentName, message);
    this.name = 'DeprecationError';
  }
}

export class AccessibilityError extends CircuitError {
  constructor(componentName: string, message: string) {
    super(componentName, message);
    this.name = 'AccessibilityError';
  }
}
