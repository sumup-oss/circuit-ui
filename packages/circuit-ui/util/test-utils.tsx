/**
 * Copyright 2020, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { FunctionComponent, ReactElement } from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import '@testing-library/jest-dom/extend-expect';
import { configureAxe } from 'jest-axe';
import { render as renderTest, RenderResult } from '@testing-library/react';
import { renderHook, act as actHook } from '@testing-library/react-hooks';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'emotion-theming';
import { light } from '@sumup/design-tokens';

import {
  ComponentsContext,
  defaultComponents,
} from '../components/ComponentsContext/ComponentsContext';

export * from '@testing-library/react';

export type RenderFn<T = any> = (component: ReactElement, ...rest: any) => T;

const WithProviders: FunctionComponent = ({ children }) => (
  <ComponentsContext.Provider value={defaultComponents}>
    <ThemeProvider theme={light}>{children}</ThemeProvider>
  </ComponentsContext.Provider>
);

const render: RenderFn<RenderResult> = (component, options) =>
  renderTest(component, { wrapper: WithProviders, ...options });
const renderToHtml: RenderFn<string> = (component) =>
  renderToStaticMarkup(<WithProviders>{component}</WithProviders>);
const create = (...args: Parameters<RenderFn<RenderResult>>) => {
  const { container } = render(...args);
  return container.children.length > 1
    ? container.children
    : container.firstChild;
};

const axe = configureAxe({
  rules: {
    // disabled landmark rules when testing isolated components.
    region: { enabled: false },
  },
});

export { create, render, renderToHtml, renderHook, actHook, userEvent, axe };
