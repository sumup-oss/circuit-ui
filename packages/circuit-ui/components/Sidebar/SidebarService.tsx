/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Children, ReactElement } from 'react';
import { SecurePayments } from '@sumup/icons';

import { isArray } from '../../util/type-check';
import { isEmpty } from '../../util/helpers';

export type Child = {
  props: {
    selected?: boolean;
    secondary?: boolean;
    visible?: boolean;
  };
};

export function hasSelectedChild(children: Child[] | Child): boolean {
  if (children) {
    const childArray = isArray(children) ? children : [children];
    const selectedChildren = childArray.filter((item) => item.props.selected);
    return !isEmpty(selectedChildren);
  }
  return false;
}

export function getSelectedChildIndex(children?: Child[] | Child): number {
  if (!children) {
    return -1;
  }

  const childArray = isArray(children) ? children : [children];

  return childArray.findIndex((child: Child) => Boolean(child.props.selected));
}

export function getSecondaryChildren(
  children?: Child[] | Child,
  visible?: boolean,
): Child[] | undefined {
  if (!children) {
    return undefined;
  }

  const childArray = isArray(children) ? children : [children];

  return childArray.map((child: Child) => ({
    ...child,
    props: { ...child.props, secondary: true, visible },
  }));
}

export function getChildrenLength(children?: Child[] | Child): number {
  if (!children) {
    return 0;
  }
  return Children.count(children);
}

export function getIcon({
  defaultIcon,
  selectedIcon,
  selected,
  disabled,
}: {
  defaultIcon?: ReactElement;
  selectedIcon?: ReactElement;
  selected?: boolean;
  disabled?: boolean;
}): ReactElement | null {
  const disabledIcon = <SecurePayments />;
  const hasIcon = !!defaultIcon;

  if (hasIcon && disabled) {
    return disabledIcon;
  }
  if (hasIcon && selectedIcon && selected) {
    return selectedIcon;
  }
  if (hasIcon) {
    return defaultIcon;
  }
  return null;
}
