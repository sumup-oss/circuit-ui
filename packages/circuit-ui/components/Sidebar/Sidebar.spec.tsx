/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { describe, expect, it, vi } from 'vitest';

import { axe, userEvent, render } from '../../util/test-utils';

import { Sidebar } from './Sidebar';

describe('Sidebar', () => {
  const baseProps = {
    children: 'Content',
    closeButtonLabel: 'Close sidebar',
    open: false,
    onClose: vi.fn(),
  };

  it('should render and match the snapshot when closed', () => {
    const props = {
      ...baseProps,
      open: false,
      onClose: vi.fn(),
    };
    const { container } = render(<Sidebar {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should render and match snapshot when open', () => {
    const props = {
      ...baseProps,
      open: true,
      onClose: vi.fn(),
    };
    const { container } = render(<Sidebar {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should dispatch onClose when CloseButton is clicked', async () => {
    const props = {
      ...baseProps,
      open: true,
      onClose: vi.fn(),
    };
    const { getByTestId } = render(<Sidebar {...props} />);
    await userEvent.click(getByTestId('sidebar-close-button'));
    expect(props.onClose).toHaveBeenCalled();
  });

  it('should dispatch onClose when the Backdrop is clicked', async () => {
    const props = {
      ...baseProps,
      open: true,
      onClose: vi.fn(),
    };
    const { getByTestId } = render(<Sidebar {...props} />);
    await userEvent.click(getByTestId('sidebar-backdrop'));
    expect(props.onClose).toHaveBeenCalled();
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const props = {
        ...baseProps,
        open: true,
        onClose: vi.fn(),
      };
      const { container } = render(<Sidebar {...props} />);
      const actual = await axe(container);
      expect(actual).toHaveNoViolations();
    });
  });
});
