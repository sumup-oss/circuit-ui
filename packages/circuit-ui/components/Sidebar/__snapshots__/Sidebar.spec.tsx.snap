// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > should render and match snapshot when open 1`] = `
.circuit-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  min-width: 256px;
  background-color: #1A1A1A;
  -webkit-transition: -webkit-transform var(--cui-transitions-default);
  transition: transform var(--cui-transitions-default);
  position: absolute;
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  z-index: var(--cui-z-index-navigation);
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

@media (min-width: 960px) {
  .circuit-0 {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    position: relative;
  }
}

@media (min-width: 960px) {
  .circuit-0 {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
  }
}

.circuit-1 {
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: #1A1A1A;
  -webkit-transition: opacity 120ms ease-in-out,visibility 120ms ease-in-out;
  transition: opacity 120ms ease-in-out,visibility 120ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  z-index: 700;
  visibility: inherit;
  opacity: 0.56;
}

@media (min-width: 960px) {
  .circuit-1 {
    visibility: hidden;
  }
}

.circuit-2 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  -webkit-transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  z-index: 800;
  visibility: inherit;
  opacity: 1;
}

@media (min-width: 960px) {
  .circuit-2 {
    visibility: hidden;
  }
}

<div>
  <nav
    class="circuit-0"
  >
    Content
  </nav>
  <div
    class="circuit-1"
    data-testid="sidebar-backdrop"
  />
  <button
    class="_base_60a0b0 _secondary_60a0b0 _m_60a0b0 _focus-visible_73b4bb _base_037c7e _m_037c7e _base_9bf092 circuit-2"
    data-testid="sidebar-close-button"
    title="Close sidebar"
    type="button"
  >
    <span
      aria-hidden="true"
      class="_loader_60a0b0"
    >
      <span
        class="_dot_60a0b0"
      />
      <span
        class="_dot_60a0b0"
      />
      <span
        class="_dot_60a0b0"
      />
      <span
        class="_hide-visually_73b4bb"
      />
    </span>
    <span
      class="_content_60a0b0"
    >
      <svg
        aria-hidden="true"
        class="_icon_60a0b0"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.71 19.29c.186.19.29.445.29.71a1 1 0 0 1-1 1c-.265 0-.52-.104-.71-.29L12 13.42l-7.29 7.29c-.19.186-.444.29-.71.29a1 1 0 0 1-1-1c0-.265.104-.52.29-.71L10.58 12 3.29 4.71A1.014 1.014 0 0 1 3 4a1 1 0 0 1 1-1c.266 0 .52.104.71.29L12 10.58l7.29-7.29c.19-.186.445-.29.71-.29a1 1 0 0 1 1 1c0 .266-.104.52-.29.71L13.42 12l7.29 7.29z"
          fill="currentColor"
        />
      </svg>
      <span
        class="_label_60a0b0"
      >
        Close sidebar
      </span>
    </span>
  </button>
</div>
`;

exports[`Sidebar > should render and match the snapshot when closed 1`] = `
.circuit-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  min-width: 256px;
  background-color: #1A1A1A;
  -webkit-transition: -webkit-transform var(--cui-transitions-default);
  transition: transform var(--cui-transitions-default);
  position: absolute;
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  z-index: var(--cui-z-index-navigation);
}

@media (min-width: 960px) {
  .circuit-0 {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    position: relative;
  }
}

.circuit-1 {
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: #1A1A1A;
  -webkit-transition: opacity 120ms ease-in-out,visibility 120ms ease-in-out;
  transition: opacity 120ms ease-in-out,visibility 120ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  z-index: 700;
}

@media (min-width: 960px) {
  .circuit-1 {
    visibility: hidden;
  }
}

.circuit-2 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  -webkit-transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  z-index: 800;
}

@media (min-width: 960px) {
  .circuit-2 {
    visibility: hidden;
  }
}

<div>
  <nav
    class="circuit-0"
  >
    Content
  </nav>
  <div
    class="circuit-1"
    data-testid="sidebar-backdrop"
  />
  <button
    class="_base_60a0b0 _secondary_60a0b0 _m_60a0b0 _focus-visible_73b4bb _base_037c7e _m_037c7e _base_9bf092 circuit-2"
    data-testid="sidebar-close-button"
    title="Close sidebar"
    type="button"
  >
    <span
      aria-hidden="true"
      class="_loader_60a0b0"
    >
      <span
        class="_dot_60a0b0"
      />
      <span
        class="_dot_60a0b0"
      />
      <span
        class="_dot_60a0b0"
      />
      <span
        class="_hide-visually_73b4bb"
      />
    </span>
    <span
      class="_content_60a0b0"
    >
      <svg
        aria-hidden="true"
        class="_icon_60a0b0"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.71 19.29c.186.19.29.445.29.71a1 1 0 0 1-1 1c-.265 0-.52-.104-.71-.29L12 13.42l-7.29 7.29c-.19.186-.444.29-.71.29a1 1 0 0 1-1-1c0-.265.104-.52.29-.71L10.58 12 3.29 4.71A1.014 1.014 0 0 1 3 4a1 1 0 0 1 1-1c.266 0 .52.104.71.29L12 10.58l7.29-7.29c.19-.186.445-.29.71-.29a1 1 0 0 1 1 1c0 .266-.104.52-.29.71L13.42 12l7.29 7.29z"
          fill="currentColor"
        />
      </svg>
      <span
        class="_label_60a0b0"
      >
        Close sidebar
      </span>
    </span>
  </button>
</div>
`;
